src/main/java/commands/DeleteCommand.java

```diff
@@ -0,0 +1,29 @@
+package commands;
+
+import exceptions.FileError;
+import storage.Storage;
+import tasklist.TaskList;
+import tasks.Task;
+import ui.Ui;
+
+/**
+ *  Handles the deletion of tasks from the user's task list.
+ */
+public class DeleteCommand extends Command {
+    private final int index;
+
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws FileError {
+        Task removedTask = tasks.getTaskList().get(index - 1);
+        tasks.getTaskList().remove(index - 1);
+        String result = "Noted. I've removed this task:\n"
+                + removedTask
+                + String.format("Now you have %d tasks in the list\n", tasks.getTaskList().size());
```

Comment:
For this portion, I would suggest starting the line with String.format. There is no need to + removedTask as that already uses the toString method. Instead doing String.format("Noted...\n%s...", removedTask.toString()...) would allow you to view the entire string as if you were reading a whole sentence or paragraph.